src/core/components/IconButton/IconButton.tsx

```diff
@@ -2,21 +2,25 @@ import IconButton from '@mui/material/IconButton';
 import { MouseEvent } from 'react';
 import './IconButton.css';
 
-interface IconButtonProps {
+interface CustomIconButtonProps {
   icon: JSX.Element;
-  label: string;
-  isSelected: boolean;
-  id: string;
+  label?: string;
+  variant?: 'primary' | 'secondary';
```

Comment:
Please define a proper enum for the variant options and use it to define it in the interface